File: go/wxpusher/message.go
package wxpusher

// private contentType 消息内容类型
type contentType int

const ContentText = contentType(1)
const ContentHtml = contentType(2)
const ContentMarkdown = contentType(3)

// Message 消息结构体
type Message struct {
	AppToken    string      `json:"appToken"`
	Content     string      `json:"content"`
	ContentType contentType `json:"contentType"`
	TopicIds    []int       `json:"topicIds"`
	UIds        []string    `json:"uids"`
	Url         string      `json:"url"`
}

func NewMessage(appToken string) *Message {
	return &Message{
		AppToken:    appToken,
		ContentType: ContentText,
	}
}

func (m *Message) SetContentType(ct contentType) *Message {
	m.ContentType = ct
	return m
}

func (m *Message) SetContent(content string) *Message {
	m.Content = content
	return m
}

func (m *Message) SetUrl(url string) *Message {
	m.Url = url
	return m
}

func (m *Message) AddUId(id string, more ...string) *Message {
	m.UIds = append(m.UIds, id)
	m.UIds = append(m.UIds, more...)
	return m
}

func (m *Message) AddTopicId(id int, more ...int) *Message {
	m.TopicIds = append(m.TopicIds, id)
	m.TopicIds = append(m.TopicIds, more...)
	return m
}

func (m Message) Send() (SendResult, error) {
	return SendMessage(m)
}

File: java/sdk/src/main/java/com/zjiecode/wxpusher/client/bean/Message.java
package com.zjiecode.wxpusher.client.bean;

import java.util.HashSet;
import java.util.Set;

/**
 * 说明：
 * 作者：zjiecode
 * 时间：2019-09-05
 */
public class Message {
    /**
     * 1:text，可以直接显示在卡片里面
     * 2:html，点击以后查看，支持html
     * 3:md，markdown格式，和html类似
     */
    public static final int CONTENT_TYPE_TEXT = 1;
    public static final int CONTENT_TYPE_HTML = 2;
    public static final int CONTENT_TYPE_MD = 3;

    private String appToken;

    //发送的目标
    private Set<String> uids;
    private Set<Long> topicIds;

    private Integer contentType;

    private String content;

    /**
     * 仅针对text消息类型有效
     */
    private String url;

    public String getAppToken() {
        return appToken;
    }

    public void setAppToken(String appToken) {
        this.appToken = appToken;
    }

    public Set<String> getUids() {
        return uids;
    }

    public void setUid(String uid) {
        this.uids = new HashSet<>(1);
        this.uids.add(uid);
    }

    public void setTopicId(Long topicId) {
        this.topicIds = new HashSet<>(1);
        this.topicIds.add(topicId);
    }

    public void setUids(Set<String> uids) {
        this.uids = uids;
    }

    public Set<Long> getTopicIds() {
        return topicIds;
    }

    public void setTopicIds(Set<Long> topicIds) {
        this.topicIds = topicIds;
    }

    public Integer getContentType() {
        return contentType;
    }

    public void setContentType(Integer contentType) {
        this.contentType = contentType;
    }

    public String getContent() {
        return content;
    }

    /**
     * 只需要 body 标签内部的内容。
     */
    public void setContent(String content) {
        this.content = content;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }
}
